servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java

```diff
@@ -124,7 +124,12 @@ protected void assignReference(NamedRef ref, String oldHash, Reference assignTo)
   @Override
   protected void deleteReference(NamedRef ref, String hash)
       throws NessieConflictException, NessieNotFoundException {
-    startAccessCheck().canDeleteReference(ref).checkAndThrow();
+    BatchAccessChecker check = startAccessCheck();
+    if (getConfig().getDefaultBranch().equals(ref.getName())) {
```

Comment:
Actually, you need to check whether `ref` is a `BranchName`

Reply:
addressed in latest commit